lib/block-supports/index.php

```diff
@@ -71,9 +73,13 @@ function gutenberg_apply_block_supports( $block_content, $block ) {
 		return $block_content;
 	}
 
+	// Structure is like `<html><head/><body/></html>`, so body is the `lastChild` of our document.
+	$body_element = $dom->documentElement->lastChild; // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+
 	$xpath      = new DOMXPath( $dom );
-	$block_root = $xpath->query( '/html/body/*' )[0];
+	$block_root = $xpath->query( './*', $body_element )[0];
```

Comment:
I had incorrectly assumed the xpath `*` query was synonymous to `DOMNode::childNodes`", but it isn't. That was causing the test failures.

The xpath `*` ["selects all element children of the context node"](https://www.w3.org/TR/1999/REC-xpath-19991116/#path-abbrev), whereas `childNodes` may be `DOMNodes` of any type, e.g. text nodes, which later error under some conditions.

I've restored the `DOMXpath` implementation to search for `*` (child elements) in the context of the `$body_element`.

An alternative I considered was to `array_search` with a function to match the `XML_ELEMENT_NODE` DOMNode::nodeType. However, I believe that would require introducing another named function which seems undesirable.

I've introduced a unit test for the text-only render output case.